src/components/tx/modals/ConfirmTxModal/ConfirmProposedTx.tsx

```diff
@@ -24,8 +25,10 @@ const ConfirmProposedTx = ({ txSummary, onSubmit }: ConfirmProposedTxProps): Rea
   const wallet = useWallet()
   const safeAddress = useSafeAddress()
   const chainId = useChainId()
+  const { page } = useTxQueue()
+
   const txId = txSummary.id
-  const canExecute = isExecutable(txSummary, wallet?.address || '')
+  const canExecute = isExecutable(txSummary, wallet?.address || '') && isNextTx(txId, page?.results || [])
```

Comment:
Can we make the `isNextTx` function part of the `isExecutable` function? Currently, `isExecutable` returns true even if a tx is not executable which seems like a bug.

Reply:
True. I'll look into it.